cli/azd/pkg/project/service_manager.go

```diff
@@ -466,7 +466,7 @@ func (sm *serviceManager) Deploy(
 
 		// Allow users to specify their own endpoints, in cases where they've configured their own front-end load balancers,
 		// reverse proxies or DNS host names outside of the service target (and prefer that to be used instead).
-		overriddenEndpoints := sm.getOverriddenEndpoints(ctx, serviceConfig)
+		overriddenEndpoints := OverriddenEndpoints(ctx, serviceConfig, sm.env)
```

Comment:
Feels like the override endpoints should just be an internal implementation detail not exposed outside of the existing `Endpoints()` function.

Reply:
agree.
It is currently baked into the Deploy() logic, after creating a service-target and deploying.

For azd show, we don't want to re-deploy to fetch the endpoints.  

I can create an issue for adding the Endpoints() method to the serviceManager, or a DeployOptions settings for asking a ServiceDeployResult w/o doing the deploy.

Reply:
Issue created: https://github.com/Azure/azure-dev/issues/2922